src/context.js

```diff
@@ -648,7 +681,12 @@ export default class Context {
 				const parsed = config != null && config.length > 0 ? JSON.parse(config) : {};
 
 				for (const key of Object.keys(parsed)) {
-					indexVariables[key] = entry;
+					const value = entry;
+					if (indexVariables[key]) {
+						const at = binarySearch(indexVariables[key], value, (a, b) => a.data.id - b.data.id);
+
+						if (at < 0) indexVariables[key].splice(-(at + 1), 0, value);
```

Comment:
I am not sure what this is doing?
The Java binary search has different semantics i.e. it encodes some meaning in the sign of the returned index, but this binary search implementation is not doing this, it is just returning -1 if it's not found. This code is probably working by coincidence, and not really generalizing correctly for all cases.
You can check the Swift code that also implements its own binary search for some inspiration.